channeld/channeld.c

```diff
@@ -993,7 +993,10 @@ static secp256k1_ecdsa_signature *calc_commitsigs(const tal_t *ctx,
 
 	msg = towire_hsm_sign_remote_commitment_tx(NULL, txs[0],
 						   &peer->channel->funding_pubkey[REMOTE],
-						   *txs[0]->input_amounts[0]);
+						   *txs[0]->input_amounts[0],
+						   (const struct witscript **) txs[0]->output_witscripts,
```

Comment:
These fields could be folded into the `towire_bitcoin_tx` and `fromwire_bitcoin_tx` helpers allowing you to just pass in `txs[0]` here and serialize/deserialize automatically.

Reply:
Might be a slightly bigger change though, so maybe we can do it in a second step :wink: